docs/advanced.rst

```diff
@@ -591,6 +591,160 @@ Discovery
 .. END STATIC DISCOVERY
 
 .. START SERVER-CLIENT DISCOVERY
+
+.. _discovery_server:
+
+Server-Client Discovery
+=======================
+
+This is based on a client-server discovery paradigm, the metatraffic (message exchange among participants to identify each
+other) is centralized in one or several server participants (right figure), as opposed to simple discovery (left
+figure), where metatraffic is exchanged using a message broadcast mechanism like an IP multicast protocol.
+
+.. image:: ds_uml.png
+    :align: center
+
+Clients must be aware of how to reach the server, usually by specifying an IP address and a transport protocol like UDP
+or TCP. Servers don't need any beforehand knowledge of their clients but, we must specify where they may be reached by
+them, usually by specifying a listening IP address and transport protocol.
+
+One of the design goals of the current implementation was to keep both the discovery messages structure and standard
+RTPS writer and reader behavior unchanged. In order to do so, clients must be aware of their server's GuidPrefix.
+GuidPrefix is the RTPS standard participant unique identifier (basically 12 bytes) which allows clients to assess
+whether they are receiving messages from the right server, as each standard RTPS message contains this piece of
+information. Note that the server's IP address may not be a reliable server's identifier because several can be
+specified and multicast addresses are acceptable. In future implementations, any other more convenient and non-standard
+identifier may substitute the GuidPrefix at the expense of adding non-standard members to the RTPS discovery messages
+structure.
+
```

Comment:
From here on, I'd restructure the entire section. I my opinion, the server-client discovery settings are:
* Discovery Protocol --> Here you'd need a table for explaining SERVER, CLIENT, and BACKUP.
* Server's GUID Prefix --> You would need to state how to specify it for both the server and the client side. Now that I think about it, I think the explanation about `GuidPrefix` fits better in the section about this.
* Server's listening address-port lists --> The uni and multicast. You'd need to explain how to set them for both server and client.
* Client's announcement period.

The other sections have a summary table with the settings, containing a short description, their value type and the default values. This is then followed by sub-sections for each of them, I think we should do the same here. Furthermore, I'd leave the specific attribute names for the code snippets. The text is explaining the concepts, so the attribute names are not needed. You can take a look at the general settings, simple discovery and static discovery for structural reference.

Reply:
But then we break with the settings or attribute centric philosophy followed in the other sections to move to a conceptual one. That looks a great idea but removes the whole 'use cases' section purpose. I'd rather stick explaining the attributes and xml tags linked with the discovery and led the people be enlightened by the tutorials.

Reply:
I would follow the outline Edu says. First a conceptual and descriptive texplanation of the client-server discovery settings and then the attributes for each of them in different subsections. At least that was the structure we agreed and as it is done for the general settings, and simple and static discovery sections.

Reply:
Ok, I'm going to think it over.